std/xml.d

```diff
@@ -2043,9 +2043,9 @@ class ElementParser
                     const startTag = startTags[tag_.name];
                     string text;
 
-                    immutable(char)* p = startTag.tagString.ptr
+                    immutable(char)* p = &startTag.tagString[0]
                         + startTag.tagString.length;
```

Comment:
>You can't do pointer arithmetic here and be @safe

Well, this whole function isn't marked as `@safe`, I'm just trying to make it more safe than it was before.

>God, this code is so ugly...

Here's a small clean-up https://github.com/dlang/phobos/pull/5380